lib/Sema/CSClosure.cpp

```diff
@@ -832,6 +832,26 @@ class SyntacticElementConstraintGenerator
   }
 
   void visitBraceStmt(BraceStmt *braceStmt) {
+    auto &ctx = cs.getASTContext();
+
+    if (auto closure = cast<ClosureExpr>(context.getAbstractClosureExpr()); context.getBody() == braceStmt) {
+      // If this closure has an empty body and no explicit result type
+      // let's bind result type to `Void` since that's the only type empty body
+      // can produce. Otherwise, if (multi-statement) closure doesn't have
+      // an explicit result (no `return` statements) let's default it to `Void`.
+      if (!constraints::hasExplicitResult(closure)){
```

Comment:
I looked at the error in the log and remembered why this check was in `generateConstraints` - `participatesInInference` called on an empty closure returns `false`, which means that the body would never be visited and this constraint would never be added. Let's try removing that check to see if it's going to break anything, otherwise we'd have to move this block back to the `generateConstraints(AnyFunctionRef, BraceStmt *)` sadly...